src/components/transition/Slider.tsx

```diff
@@ -1,7 +1,17 @@
 import { useCallback, useEffect, useRef, useState } from 'preact/hooks';
+import type { MutableRef } from 'preact/hooks';
 
 import type { TransitionComponent } from '../../types';
 
+function isTransitionRelevant(
+  e: TransitionEvent,
+  containerRef: MutableRef<HTMLDivElement | null>
+): boolean {
+  // Only transitions on the actual component and for the "height" property are
+  // relevant "in"/"out" transitions.
+  return e.target === containerRef.current && e.propertyName === 'height';
+}
+
```

Comment:
Are you checking on the `height` property to be thorough, or was there a specific case that you can conceive of where this would matter? Are you thinking of a world where we'd apply more than one Transition to a component that takes a `TransitionComponent`? I can think of some _possible_ cases where checking the property might matter, but I'm not clear on whether it's meaningfully applicable to our real life.

I'm wondering if this might be simplified down to check for the `e.target` (only) and a simpler (commented) conditional within `handleTransitionEnd` (versus being its own function).

Reply:
> Are you checking on the height property to be thorough, or was there a specific case that you can conceive of where this would matter?

To be more explicit on what properties are relevant for the Slider's in->out or out->in change. We might at some point add other transitions, like maybe we want to change the background color on hover, and that should not trigger `onTransitionEnd`.

> I'm wondering if this might be simplified [...] a simpler (commented) conditional within handleTransitionEnd (versus being its own function).

That was the first iteration of this, and I'm thinking it probably makes sense.

I wrapped it on its own function to give it a descriptive name, but the name is actually not very descriptive on the why, and the comment inside is actually in the wrong place and should be in `handleTransitionEnd` instead.